PMNS_LIV.cxx

```diff
@@ -169,24 +138,30 @@ void PMNS_LIV::SetcT(int flvi, int flvj, double val, double phase){
     cout << " neutrinos. Doing nothing." << endl;
     return;
   }
+  if(dim != 4 && dim != 6 && dim != 8){
+    cout << "cT coefficient dimension should be an even value, starting from 4\n";
+    return;
+  }
 
-  complexD h = val;
+  int pos = dim == 4 ? 0: dim == 6 ? 1 : dim == 8 ? 2 : 4;
+  complexD h[3];
+  h[pos] = val;
```

Comment:
There's no need to create 3 temporary variables here. You will only ever use one in this function.